swan-cern/files/swan_config_cern.py

```diff
@@ -39,34 +56,29 @@ def _init_eos_secret(self):
         except Exception as e:
             raise ValueError("Could not create required user credential")
 
-
         # ITHADOOP-819 - Ports need to be opened using service creation, and later assigning allocated service nodeport to a pod 
         # Create V1Secret with eos token
-        try:
-            secret_data = client.V1Secret()
-
-            secret_meta = client.V1ObjectMeta()
-            secret_meta.name = eos_secret_name
-            secret_meta.namespace = swan_container_namespace
-            secret_meta.labels = {
-                "swan_user": username
-            }
-            secret_data.metadata = secret_meta
-            secret_data.data = {}
-            secret_data.data['krb5cc'] = eos_token_base64
+        secret_data = V1Secret()
+
+        secret_meta = V1ObjectMeta()
+        secret_meta.name = eos_secret_name
+        secret_meta.namespace = swan_container_namespace
+        secret_meta.labels = {
+            "swan_user": username
+        }
+        secret_data.metadata = secret_meta
+        secret_data.data = {}
+        secret_data.data['krb5cc'] = eos_token_base64
 
+        try:
+            # eos-tokens secret is cleaned when user session ends, so try creating it
+            await self.spawner.api.create_namespaced_secret(swan_container_namespace, secret_data)
+        except ApiException:
+            # A secret with the same name exists, probably a remnant of a wrongly-terminated session, then replace it
```

Comment:
Do we have a log object? This should be logged as a warning (since even wrongly terminated sessions should still remove it, no?)

Reply:
Wrongly terminated sessions should remove it, yes, since the removal code that is part of `stop` is in a `finally` block. I guess it could happen that the hub crashes and the `stop` method is never called for the user session. I can add the warning anyway, though, we should have access to the logger of the spawner from here.